From 20893b91210566ed881ce186c2531b8ac30c983e Mon Sep 17 00:00:00 2001
From: Kristian Rietveld <kris@lanedo.com>
Date: Thu, 4 Oct 2012 09:06:01 +0200
Subject: [PATCH 21/21] quartz: Actually use the window background PATTERN
 color

Before we used a window's background color, which resulted in corrupted
display in some cases, presumably because we didn't reset the active
pattern. This patch seems to eliminate the observed corruption.
---
 gdk/quartz/GdkQuartzView.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gdk/quartz/GdkQuartzView.c b/gdk/quartz/GdkQuartzView.c
index 56307b9..dd7b998 100644
--- a/gdk/quartz/GdkQuartzView.c
+++ b/gdk/quartz/GdkQuartzView.c
@@ -97,7 +97,7 @@
        */
       [NSGraphicsContext saveGraphicsState];
 
-      [[[self window] backgroundColor] setFill];
+      [[NSColor windowBackgroundColor] setFill];
       [NSBezierPath fillRect:rect];
 
       [NSGraphicsContext restoreGraphicsState];
-- 
1.7.5.4

